File: C#/Async.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;

class Scroller
{
    static async Task<long> GetWebPageSizeAsync(string url)
    {
        try
        {
            using HttpClient client = new();
            // Send the request and get the response
            HttpResponseMessage response = await client.GetAsync(url);
            response.EnsureSuccessStatusCode(); // Throws if not successful

            // Check Content-Length header
            if (response.Content.Headers.ContentLength.HasValue)
            {
                return response.Content.Headers.ContentLength.Value;
            }
            else
            {
                // Fallback: read the content
                byte[] content = await response.Content.ReadAsByteArrayAsync();
                return content.Length;
            }
        }
        catch (Exception ex)
        {
            Console.WriteLine($"Error: {ex.Message}");
            return -1;
        }
    }

    static async Task Main()
    {
        string url = "https://example.com";
        long size = await GetWebPageSizeAsync(url);
        Console.WriteLine($"Size of webpage '{url}': {size} bytes");
    }
}

File: Py/async.py
import aiohttp
import asyncio


async def get_webpage_size(url):
    try:
        async with aiohttp.ClientSession() as session:
            async with session.get(url) as response:
                # Get the content length from headers if available
                content_length = response.headers.get('Content-Length')
                if content_length:
                    size = int(content_length)
                else:
                    # If not available, read the full content
                    content = await response.read()
                    size = len(content)
                return size
    except Exception as e:
        print(f"Error: {e}")
        return -1

async def main():
    url = "https://example.com"
    size = await get_webpage_size(url)
    print(f"Size of webpage '{url}': {size} bytes")

# Run the async function
if __name__ == "__main__":
    asyncio.run(main())